view/infinite.go

```diff
@@ -0,0 +1,158 @@
+	measurement := &res.Results[0]
+	if len(ctx.Stats) == 0 {
+		// Initialize state
+		ctx.Stats = make([]model.MeasurementStats, 1)
+		// Print header
+		fmt.Println(generateHeader(measurement, !ctx.CI))
+		fmt.Printf("PING %s (%s) 56(84) bytes of data.\n", res.Target, measurement.Result.ResolvedAddress)
+	}
+	timings, err := client.DecodePingTimings(measurement.Result.TimingsRaw)
+	if err != nil {
+		return err
+	}
+	for i := range timings {
+		ctx.Stats[0].Sent++
+		t := timings[i]
+		fmt.Printf("64 bytes from %s (%s): icmp_seq=%d ttl=%d time=%.2f ms\n",
+			measurement.Result.ResolvedHostname,
+			measurement.Result.ResolvedAddress,
+			ctx.Stats[0].Sent,
+			t.TTL,
+			t.RTT)
+	}
+	return nil
+}
+
+func outputMultipleLocations(res *model.GetMeasurement, ctx *model.Context) error {
+	var err error
+	if len(ctx.Stats) == 0 {
+		// Initialize state
+		ctx.Stats = make([]model.MeasurementStats, len(res.Results))
+		for i := range ctx.Stats {
+			ctx.Stats[i].Min = math.MaxFloat64
+		}
+		// Create new writer
+		ctx.Area, err = pterm.DefaultArea.Start()
+		if err != nil {
+			return errors.New("failed to start writer: " + err.Error())
+		}
+	}
+	tableData := pterm.TableData{
+		{"Location", "Loss", "Sent", "Last", "Avg", "Min", "Max"},
+	}
+	for i := range res.Results {
+		result := &res.Results[i]
+		localStats := &ctx.Stats[i]
+		updateMeasurementStats(localStats, result)
+		tableData = append(tableData, []string{
+			getLocationText(result),
+			formatValue(fmt.Sprintf("%.2f", localStats.Loss)+"%", 6),
+			formatValue(fmt.Sprintf("%d", localStats.Sent), 3),
+			formatValue(formatDuration(localStats.Last), 7),
+			formatValue(formatDuration(localStats.Avg), 7),
+			formatValue(formatDuration(localStats.Min), 7),
+			formatValue(formatDuration(localStats.Max), 7),
+		})
+	}
+	t, err := pterm.DefaultTable.WithHasHeader().WithData(tableData).Srender()
+	if err != nil {
+		return err
+	}
+	ctx.Area.Update(t)
+
+	return nil
+}
+
+func formatDuration(ms float64) string {
+	if ms < 10 {
+		return fmt.Sprintf("%.2f ms", ms)
+	}
+	if ms < 100 {
+		return fmt.Sprintf("%.1f ms", ms)
+	}
+	return fmt.Sprintf("%.0f ms", ms)
+}
+
+func formatValue(v string, width int) string {
+	for len(v) < width {
+		v = " " + v
+	}
+	return pterm.NewStyle(pterm.FgDefault).Sprint(v)
+}
+
+func updateMeasurementStats(localStats *model.MeasurementStats, result *model.MeasurementResponse) error {
+	stats, err := client.DecodePingStats(result.Result.StatsRaw)
+	if err != nil {
+		return err
+	}
+	timings, err := client.DecodePingTimings(result.Result.TimingsRaw)
+	if err != nil {
+		return err
+	}
+	if stats.Min < localStats.Min && stats.Min != 0 {
+		localStats.Min = stats.Min
+	}
+	if stats.Max > localStats.Max {
+		localStats.Max = stats.Max
+	}
+	if stats.Avg != 0 {
+		localStats.Avg = (localStats.Avg*float64(localStats.Sent) + stats.Avg*float64(stats.Total)) / float64(localStats.Sent+stats.Total)
+	}
```

Comment:
There seems to be an issue when the loss is 100% (try 1.9.9.9 as target)
https://dl.dropboxusercontent.com/scl/fi/yvsef3j6vn40f51216mnm/2024-01-19_17-05-13.png?rlkey=2qyecrapvpla0u2v21u0uq6f5&dl=0